lib/clockwise_import.rb

```diff
@@ -0,0 +1,118 @@
+require 'csv'
+
+class ClockwiseImporter
+  def initialize(csv_path:, tenant:)
+    @csv_path = csv_path
+    @tenant = tenant
+  end
+
+  def import!
+    Apartment::Tenant.switch @tenant
+    Entry.transaction do
+      CSV.foreach(@csv_path,
+                  headers: true,
+                  quote_char: '"',
+                  col_sep: ';') do |r|
+        row = ImportRow.new(r)
+        EntryBuilder.new(row).create! unless row.hours < 1
+      end
+    end
+  end
+end
+
+class EntryBuilder
+  def initialize(row)
+    @row = row
+  end
+
+  def create!
+    client = build_client
+    project = build_project(client)
+    category = build_category
+    user = build_user
+    Entry.create!(
+      project: project,
+      category: category,
+      user: user,
+      hours: @row.hours,
+      date: @row.date,
+      description: @row.description,
+    )
+  end
+
+  private
+
+  def build_client
+    Client.where('lower(name) = ?', @row.client)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.